bpo-35345: Remove platform.popen()

vstinner opened the pull request:
Remove platform.popen() function, it was deprecated since Python 3.3:
use os.popen() instead.

<!-- issue-number: [bpo-35345](https://bugs.python.org/issue35345) -->
https://bugs.python.org/issue35345
<!-- /issue-number -->

vstinner committed:
bpo-35345: Remove platform.popen()

Remove platform.popen() function, it was deprecated since Python 3.3:
use os.popen() instead.

vstinner committed:
Fix reST syntax in the doc

serhiy-storchaka reviewed (commented):
LGTM except What's New entries. This is not new feature, and is not changing behavior of existing feature. I think the only entry should be added into the "Removed" section.

vstinner committed:
Merge branch 'master' into remove_platform_popen

vstinner committed:
Move What's New entry to Removed section

Rename also the "Removed" section to "API and Feature Removals": copy
title from What's New in Python 3.7.
